### ISSUE TITLE
Focus on datepicker footer triggers extra selects

### ISSUE THREAD

**Issue by Nosfistis** [ng-bootstrap/ng-bootstrap#3444]
### Bug description:
When using a datepicker with a custom footer, if that footer contains focusable elements (e.g. inputs), navigating between those elements with keyboard tab causes select events which are invalid. The selection payload is not an `NgbDate` instance, but a JS `Event` with `type: 'select'`.

### Link to minimally-working StackBlitz that reproduces the issue:

https://stackblitz.com/edit/ng-bootstrap-footer-focus

### Versions of Angular, ng-bootstrap and Bootstrap:

Angular: 8.2.12

ng-bootstrap: 5.1.2

Bootstrap: 4.3.0

**Comment by fbasso** [comment]
Thanks for opening this your issue @Nosfistis .

After a quick investigation, it seems it's because 'select' is a native event which is triggered by the browser when the input is focused, which is the same name of the one used to select a day. It is not supported by all the browsers.

https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onselect

**Comment by Nosfistis** [comment]
I did not think of a native event at all!

It confused my why it's not named  `(dateSelect)`, same as the directive one. As a workaround. I am using a filter on the event, checking if it returns payload of type `NgbDate`.

**Comment by maxokorokov** [comment]
Hey, @Nosfistis. Thanks for reporting this.

- `Datepicker` was created before the `InputDatepicker`
- `Datepicker` was never could raise `select` event, so the output was named `(select)`
- `InputDatepicker` was added on the `<input>` element, so we had to name similar output `(dateSelect)`
- then we added a footer to the `Datepicker` and got this issue :)

---

Anyway I suggest we:
1. introduce `(dateSelect)` to the datepicker
2. deprecate `(select)`
3. remove `(select)` in the next major release
